self-core-impl/src/test/java/com/selfxdsd/core/GitlabRepoTestCase.java

```diff
@@ -159,5 +160,39 @@ public void returnsLabels() {
         );
     }
 
-
+    /**
+     * A GitlabRepo can return its Stars.
+     */
+    @Test
+    public void returnsStars() {
+        MockJsonResources resources = new MockJsonResources(
+            req -> {
+                return new MockJsonResources.MockResource(
+                    HttpURLConnection.HTTP_OK,
+                    Json.createObjectBuilder().add(
+                        "path_with_namespace",
+                        "test/repo"
+                    ).build()
+                );
+            }
+        );
+        final Repo repo = new GitlabRepo(
+            resources,
+            URI.create("https://gitlab.com/projects/test/repo"),
+            Mockito.mock(User.class),
+            Mockito.mock(Storage.class)
+        );
+        MatcherAssert.assertThat(
+            repo.stars(),
+            Matchers.allOf(
+                Matchers.notNullValue(),
+                Matchers.instanceOf(GitlabStars.class)
+            )
+        );
+        repo.stars().add();
+        MatcherAssert.assertThat(
+            resources.requests().last().getUri().toString(),
+            Matchers.equalTo("https://gitlab.com/projects/test/repo/star")
+        );
```

Comment:
@fellahi-ali You can remove this assert and ``repo.starts().add()``. In this test, we are only interested to see if GitlabRepo returns the proper instance of Stars.

Reply:
@amihaiemil Sure! actually I felt I'm overthinking there :(